src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -152,7 +152,7 @@ public string ConfigFilePath
                 {
                     validSettingFiles.AddRange(
                         GetSettingsFileNames(root)
-                            .Select(f => ReadSettings(root, f))
+                            .Select(f => ReadSettings(root, f, isMachineWideSettings: false, throwOnFailure: false))
```

Comment:
why not changing the default value of throwOnFailure to false and just set it to true where we need it?